src/mixins/actionGlobal.js

```diff
@@ -19,14 +19,14 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-import Vue from 'vue'
+import { warn } from 'vue'
 
 export default {
 	before() {
 		// all actions requires a valid text content
 		// if none, forbid the component mount and throw error
-		if (!this.$slots.default || this.text.trim() === '') {
-			Vue.util.warn(`${this.$options.name} cannot be empty and requires a meaningful text content`, this)
+		if (!this.$slots.default() || this.text.trim() === '') {
+			warn(`${this.$options.name} cannot be empty and requires a meaningful text content`, this)
```

Comment:
How does this part work? o_O

Reply:
What exactly do you mean? I didn't write it, originally, I only adjusted it for vue 3.

Reply:
Sorry, I missed that it was an old part. I just don't understand how it works in general, who and where call this `before()` custom option.

Reply:
Yes, I also find it a bit weird. Every `Action*` component, e.g. `ActionButton` uses this mixin to check that either `text` or the default slot are present.